packages/affiliates/libs/uniswap/models.ts

```diff
@@ -0,0 +1,253 @@
+import assert from "assert";
+import { exists, getPositionKey } from "./utils";
+import { ethers } from "ethers";
+
+type CacheType = {
+  [key: string]: string | number | boolean;
+};
+export const Cache = (table = new Map<string | number, CacheType>()) => {
+  function list() {
+    return [...table.values()];
+  }
+  return {
+    has: table.has.bind(table),
```

Comment:
I wasn't sure exactly how `bind` works but found this article about how `bind` loses some type safety, maybe just use slightly more verbose arrow functions instead? https://hamednourhani.gitbooks.io/typescript-book/content/docs/tips/bind.html

Reply:
interesting, yea sounds good